tests/unit-tests/test_nn_model.py

```diff
@@ -932,11 +1042,105 @@ def test_hgt_with_edge_features(input_dim, output_dim, dev):
+        layerwithef.eval()
+
+        edge_feats = {
+            ("n0", "r0", "n1"): th.rand(r0_eid.shape[0], input_dim).to(dev)
+        }
+
+        emb = layerwithef(block, node_feats, edge_feats)
+        
+        assert 'n0' not in emb
+        assert emb['n1'].shape[0] == len(seeds['n1'])
+        assert emb['n1'].shape[1] == output_dim 
+
+        # r1 case
+        layerwithef = HGTLayerwithEdgeFeat(
+            input_dim, output_dim, 
+            ntypes, etypes,
+            num_heads=4,
+            edge_feat_name={("n0", "r1", "n1"): ['feat']},
+            edge_feat_mp_op=e_f_mp_op,
+            activation=th.nn.ReLU(),
+            dropout=0.0,
+            norm='')
+        init_hgtlayer(layerwithef)
+        layerwithef = layerwithef.to(dev)
+        layerwithef.eval()
+
+        edge_feats = {
+            ("n0", "r1", "n1"): th.rand(r1_eid.shape[0], input_dim).to(dev)
+        }
+
+        emb = layerwithef(block, node_feats, edge_feats)
+        
+        assert 'n0' not in emb
+        assert emb['n1'].shape[0] == len(seeds['n1'])
+        assert emb['n1'].shape[1] == output_dim
+    
+        # Test case 3: abnormal case, initialize HGTLayerwithEdgeFeat layer with no edge feature name.
+        #               this will trigger an assertion error of empty edge 
+        with assert_raises(AssertionError):
+            layerwithef = HGTLayerwithEdgeFeat(
+                input_dim, output_dim, 
+                ntypes, etypes,
+                num_heads=4,
+                edge_feat_mp_op=e_f_mp_op,
+                activation=th.nn.ReLU(),
+                dropout=0.0,
+                norm='')
+
+        # Test case 4: abnormal case, initialize HGTLayer layer with edge feature name.
+        #               this will trigger a type error of unexpected keyword argument
+        with assert_raises(TypeError):
+            layerwithef = HGTLayer(
+                input_dim, output_dim, 
+                ntypes, etypes,
+                num_heads=4,
+                edge_feat_name={("n0", "r1", "n1"): ['feat']},
+                edge_feat_mp_op=e_f_mp_op,
+                activation=th.nn.ReLU(),
+                dropout=0.0,
+                norm='')
+
+        # Test case 5: abnormal case, layer has edge feature weights, but give no edge features.
+        #               this will trigger an assertion error of no edge features provided for message.
+        layerwithef = HGTLayerwithEdgeFeat(
+            input_dim, output_dim, 
+            ntypes, etypes,
+            num_heads=4,
+            edge_feat_name={("n0", "r0", "n1"): ['feat'], ("n0", "r1", "n1"): ['feat']},
+            edge_feat_mp_op=e_f_mp_op,
+            activation=th.nn.ReLU(),
+            dropout=0.0,
+            norm='')
+        init_hgtlayer(layerwithef)
+        layerwithef = layerwithef.to(dev)
+        layerwithef.eval()
+
+        with assert_raises(AssertionError):
+            layerwithef(block, node_feats)
 
```

Comment:
Could you add another abnormal case, where initialize `HGTLayerwithEdgeFeat` with edge feature name, but provide an empty dict as edge feature in forward(). This will trigger an assertion error of no edge feature provided for MP.